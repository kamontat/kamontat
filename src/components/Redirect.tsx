import React from "react"
import { Helmet } from "react-helmet"
import { BaseOptions } from "../layout/Base"

import { Meta } from "./Metadata"

interface RedirectOptions extends BaseOptions {
  link?: string
  second?: number
}

const Redirect = ({ link, second }: RedirectOptions): JSX.Element => (
  <Helmet
    meta={[{ "http-equiv": "refresh", content: `${second ?? 0}; url=${link ?? "/?error=true"}` } as Meta]}
  ></Helmet>
)

export default Redirect
