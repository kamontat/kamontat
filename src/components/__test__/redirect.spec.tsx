import React from "react"
import { render, waitFor } from "@testing-library/react"
import Redirect from "../Redirect"

describe("Redirect", () => {
  it("without redirect link specify", async () => {
    render(<Redirect />)

    let data: HTMLElementTagNameMap["meta"] | undefined = undefined
    await waitFor(() => {
      expect(document.querySelector("meta")?.getAttribute("http-equiv")).toEqual("refresh")

      data = document.querySelector("meta") ?? undefined
    })

    expect(data).toMatchSnapshot()
  })

  it("with redirect link and default delay", async () => {
    render(<Redirect link="https://google.com" />)

    let data: HTMLElementTagNameMap["meta"] | undefined = undefined
    await waitFor(() => {
      expect(document.querySelector("meta")?.getAttribute("http-equiv")).toEqual("refresh")
      data = document.querySelector("meta") ?? undefined
    })

    expect(data).toMatchSnapshot()
  })

  it("with redirect link and custom delay", async () => {
    render(<Redirect link="https://google.com" second={30} />)

    let data: HTMLElementTagNameMap["meta"] | undefined = undefined
    await waitFor(() => {
      expect(document.querySelector("meta")?.getAttribute("http-equiv")).toEqual("refresh")
      data = document.querySelector("meta") ?? undefined
    })

    expect(data).toMatchSnapshot()
  })
})
