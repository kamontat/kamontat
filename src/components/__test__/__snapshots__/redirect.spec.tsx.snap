// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redirect with redirect link and custom delay 1`] = `
<meta
  content="30; url=https://google.com"
  data-react-helmet="true"
  http-equiv="refresh"
/>
`;

exports[`Redirect with redirect link and default delay 1`] = `
<meta
  content="0; url=https://google.com"
  data-react-helmet="true"
  http-equiv="refresh"
/>
`;

exports[`Redirect without redirect link specify 1`] = `
<meta
  content="0; url=/?redirect=failed"
  data-react-helmet="true"
  http-equiv="refresh"
/>
`;
